src/brevitas/graph/equalize.py

```diff
@@ -1384,27 +1389,44 @@ def _apply_rotate(
 
         for name, indexes in region.sinks.items():
             module = region.get_module_from_name(name)
+            weight_axis = _get_input_axis(module)
+
             # Only "weight" is rotated
-            tensor_names_axis = [("weight", _get_input_axis(module))]
+            if region.expand_region:
+                assert isinstance(module, nn.Linear), "Currently only Linear layers support expanded hadamard"
```

Comment:
The code just below can support any layer type, the issue is more on the activation side since we need to define also the dimension along which the input needs padding. It is not complicated to do tbh, but not necessary atm